internal/cortex/frontend/transport/handler.go

```diff
@@ -161,6 +163,16 @@ func (f *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// isQueryEndpoint returns true if the path is any of the Prometheus HTTP API,
+// query-related endpoints.
+// Example: /api/v1/query, /api/v1/query_range, /api/v1/series, /api/v1/label, /api/v1/labels
+func isQueryEndpoint(path string) bool {
+	return strings.HasPrefix(path, "/api/v1/query") ||
```

Comment:
Should we just check for `api/v1`?

Reply:
There is still a bunch of API endpoints that have nothing to do with querying, but we can go that route as well.